### PR TITLE
Migrate all facter variables to new array structure

### PR DESCRIPTION
**Pull request by ppp0**
Starting with puppet4 fact variables are looked up as element out of the $facts array
 
E.g. `$::facts['fdqn']` instead of ~~$::fqdn~~

### CODE REVIEW

File: modules/monit/templates/monitrc [hunk 1]
```diff
@@ -4,7 +4,7 @@ set logfile syslog facility log_daemon
 set mailserver localhost
 set mail-format {
 	from: <%= @emailFromWithDefault %>
-	subject: monit $ACTION - $EVENT: $SERVICE on <%= @clientcert %>
+	subject: monit $ACTION - $EVENT: $SERVICE on <%= @trusted['clientcert'] %>
```
Comment (kris-lab) [inline modules/monit/templates/monitrc]: `clientcert` is in `$::facts`... `certname` is in `@::trusted`?

**Comment by kris-lab** [comment]
@ppp0 I have added one comment

**Comment by njam** [comment]
The build is still running - is it stuck? http://ci.cargomedia.ch:8080/job/cargomedia-puppet-packages-pr/190/console

**Comment by ppp0** [comment]
retest this please

**Comment by ppp0** [comment]
retest this please

**Comment by ppp0** [comment]
@kris-lab as discussed, I merged the changes made in #1347 (remove Wheezy support for fluentd-related tests) and fixed the other failing specs. I believe it should pass now. Feel free to merge this!

**Comment by ppp0** [comment]
retest this please

**Comment by kris-lab** [comment]
ok, please merge in the right order and test

**Comment by ppp0** [comment]
All right I fixed the failing specs:
* `fluentd` not starting: all plugins need to be there as well, so we can't `require 'fluentd'`. It would start without the plugins being there (not always, timing problem)
* Trusted domain: on a ubuntu vivid vagrant box, the domain is set to `bsl.cargomedia.net`..

**Comment by ppp0** [comment]
paired with @kris-lab

**Pull request merged**